docs/vnext/contract.md

```diff
@@ -0,0 +1,70 @@
+# Azure Deployments Extensibility Provider API Contract
+
+This document defines the contract of the API that every team who wants to onboard Azure Deployments extensibility provider must adhere to. An Azure Deployments extensibility provider is an API abstraction, allowing users to do the following operations:
+1. Deploy Azure data-plane or non-Azure resources (a.k.a, Azure Deployments extensible resources) via Bicep files or ARM templates;
+2. [Future Plan] Perform resource-level or provider-level queries that have no side effects through provider functions in Bicep files or ARM templates, e.g., list secrets of a resource.
+
+## Table of Contents
+- [Api Limits and Constraints](#api-limits-and-constraints)
+- [Resource API Reference](#resource-api-reference)
+- [Long-running Operation Reference](#long-running-operation-reference)
+
+## API Limits and Constraints
+
+### Request Timeout
+
+The timeout value for each request is 60 seconds. If the provider failed to respond within 60 seconds, the Extensibility Host will return a 504 Gateway Timeout to the deployments RP and ignore the response from the provider.
+
+### Request Throttling
+
+The Extensibility Host does not implement throttling for incoming request from the deployments RP, and it's up to extensibility providers to apply their own throttling limits. However, since the extensible resources are deployed through ARM, the [general ARM request throttling limits](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/request-limits-and-throttling) still apply.
+
+### Maximum Request Body Size
+
+> This is something that the deployments team has been trying to improve. The limit may change in the future.
+
+The maximum size of a request body that ARM will accept is 4 MB. Any request with a body larger than 4 MB will not be sent to the extensibility host and providers, and a 413 Payload Too Large will be returned to the client.
+
+### Maximum Response Size
+
+To keep it consistent with ARM's limit, the extensibility host does not accept any single resource response that exceeds 20MB in size. When this happens, the response body will be dropped and a 502 Bad Gateway error will be returned to the deployments RP.
+
+## Resource API Reference
+
+For details about resource APIs, Refer to the OpenAPI 3.0 based resource API [specification](openapi.yaml). You may view it in the ReDoc UI [here](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/Azure/bicep-extensibility/master/docs/vnext/openapi.yaml) for better experience.

```

Comment:
View the OpenAPI spec on the PR branch: https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/Azure/bicep-extensibility/shenglol/vnext-spec/docs/vnext/openapi.yaml&nocors